sdk/core/Azure.Core/src/Shared/RetriableStream.cs

```diff
@@ -198,6 +228,7 @@ public override void Flush()
 
             protected override void Dispose(bool disposing)
             {
+                _onDispose?.Invoke();
```

Comment:
- If `_onDispose` throws an exception, should we continue disposing the stream?
- If `_onDispose` recursively calls stream dispose, what should happen in this case?
- Do we care about `disposing` parameter?

Reply:
@AlexanderSher, these are great questions.  I know you've recently done some work with the dispose pattern, would you be willing to make a recommendation for the preferred approach in this case?

Reply:
Per offline conversation, using a new approach suggested by @AlexanderSher.